crates/matrix-sdk-base/src/rooms/mod.rs

```diff
@@ -141,6 +145,9 @@ impl BaseRoomInfo {
     /// Returns true if the event modified the info, false otherwise.
     pub fn handle_state_event(&mut self, ev: &AnySyncStateEvent) -> bool {
         match ev {
+            AnySyncStateEvent::BeaconInfo(b) => {
+                self.beacons.insert(b.state_key().clone(), b.into());
```

Comment:
Aren't beacon infos ever removed/expired?

Reply:
`beacon_info`'s are never removed -- but they can expire.

Would you like to see the handling of expiring `beacon_info`'s in this MR or in a follow up?

Reply:
Followup's fine to me!